client/transaction-pool/src/lib.rs

```diff
@@ -54,6 +60,57 @@ pub struct BasicPool<PoolApi, Block>
 	api: Arc<PoolApi>,
 	revalidation_strategy: Arc<Mutex<RevalidationStrategy<NumberFor<Block>>>>,
 	revalidation_queue: Arc<revalidation::RevalidationQueue<PoolApi>>,
+	ready_poll: Arc<Mutex<ReadyPoll<ReadyIteratorFor<PoolApi>, Block>>>,
+}
+
+struct ReadyPoll<T, Block>
+where
+	Block: BlockT,
+{
+	updated_at: Option<NumberFor<Block>>,
+	pollers: Vec<(NumberFor<Block>, oneshot::Sender<T>)>,
+}
+
+impl<T, Block> Default for ReadyPoll<T, Block>
+where
+	Block: BlockT,
+{
+	fn default() -> Self {
+		Self {
+			updated_at: None,
+			pollers: Default::default(),
+		}
+	}
+}
+
+impl<T, Block> ReadyPoll<T, Block>
+where
+	Block: BlockT,
+{
+	fn trigger(&mut self, number: NumberFor<Block>, iterator_factory: impl Fn() -> T) {
+		self.updated_at = Some(number);
+
+		let pollers = self.pollers.drain(..).collect::<Vec<(_, _)>>();
```

Comment:
I don't like that we allocate each time we call this function :( 

```
let mut i = 0;
while i < self.pollers.len() {
    if self.pollers[i].0 <= number {
        let poller_sender = self.pollers.swap_remove(i);
				log::debug!(target: "txpool", "Sending ready signal at block {}", number);
				let _s = poller_sender.send(iterator_factory());
    } else {
        i += 1;
    }
}
```

Reply:
This would not require the extra allocation.

Reply:
I would've bothered if this happened more often than once per block

will update